rfcs/text/0000-rfc-application-layer-bundle-types-and-definitions.md

```diff
@@ -0,0 +1,431 @@
+### DCP PR:
+
+Link to document view: [here](https://github.com/HumanCellAtlas/dcp-community/blob/d2da58b16efd80fa3e31ed2194169efd79b02aa2/rfcs/text/0000-rfc-application-layer-bundle-types-and-definitions.md)
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# Application Layer Bundle Types and Definitions
```

Comment:
I find the title of this RFC confusing, as I don't know what the application layer and have not seen it as part of any other DCP documentation.  Is this the data browser?   the API?

I think it would be better to not bring in terminology that isn't currently part of the mental DCP lexicon.

Although, we do need a written DCP lexicon!!

Reply:
Reading wikipedia: [Application layer](https://www.wikiwand.com/en/Application_layer), the title makes even less sense.

Reply:
I mean "Application Layer" in the multi tier architecture sense https://en.wikipedia.org/wiki/Multitier_architecture

Not the OSI Layer 7 networking protocol layer sense.

<img width="975" alt="Screen Shot 2019-10-01 at 10 44 28 AM" src="https://user-images.githubusercontent.com/3239697/65986412-a9d32780-e438-11e9-91fe-726a909e7de2.png">

Reply:
I have added a note to the top of the doc that should clear the issue about the "Application Layer" term.

>Note: I have used the term "Application Layer" extensively in this document and the term seems to be causing some confusion. Let me clarify that I mean "Application Layer in the Multi Tier Architecture sense as opposed to the "Data Layer". Here the "Appplications" are the boxes or clients of DSS. Whoever writes into the database (DSS) needs to validate the records (bundles) they create are appropriately shaped as the database (DSS) does not to this.

I will resolve all comments about the "Application Layer" term. Please reopen if you feel this is not clear.

Reply:
It is still very confusing as the terminology is not used in the DCP and I still don't know what software comprises the application layer.  Azul?

Reply:
I still feel that "application layer" is just introducing confusion from another programming model that is not used in the DCP.  "DSS client" or "DSS application" is clearer in reference to the DSS.  In terms of the overall DCP,

Maybe this is something to bring up in tech arch???

Reply:
p.s.  I do see the distinction trying to be made here and I think it is important, it is only the term that seems problematic.